src/network/networkd-link.c

```diff
@@ -689,7 +689,9 @@ static Address* link_find_dhcp_server_address(Link *link) {
 static int link_enter_configured(Link *link) {
         assert(link);
         assert(link->network);
-        assert(link->state == LINK_STATE_SETTING_ROUTES);
+
+        if (link->state != LINK_STATE_SETTING_ROUTES)
+                return 1;
```

Comment:
`link_down_handler` is a callback, and for callbacks `1` means break look, and `0` means continue loop. That's a different case.

You could actually make `link_enter_configured` return `void`.